_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+  // The tree of components that this component represents
+  render() {
+    // This is JSX code, JSX is a source-code transformer that converts code from 
+    // HTML-like brackets into a specific method call. E.g  `<Text font="Garamond" />`
+    // turns into; `React.createElement('Text', {font: 'Garamond'}, null)`
+    return (
+      <View>
+        <TextInput text={{ value: props.query }} searching={props.searching} />
+        <ScrollView>
+          {props.results.map(rowForResult)}
+        </ScrollView>
+      </View>
+    )
+  }
+  // Returns a single component for a row
+  rowForResult(result) {
+    return (
+      <Result>
+        <Image source={{ uri: result.url }} />
+        <Text>{result.name}</Text>
+      </Result>
+    )
+  }
+}
+```
+{% endraw %}
+
+> You're looking at a subclass of `React.Component` with two functions, `render` and `rowForResult`. `render` is the key function for defining your tree.
+
+Instead of MVC, React uses composition of components to handle complexity - this should feel quite similar to iOS development. The screen of an iOS app is typically made up of `UIView`s, and `UIViewController`s which exist as 2 interlinked trees of hierarchy. A `UIViewController` itself doesn't have a visual representation, but exists to manipulate data, handle actions and the view structure for `UIViews` who do. 
+
+By merging the responsibilities of a `UIView` and `UIViewController` into a Component, there is a consistent way to work with all aspects of your app. 
+
+To try to undestand this, let's take a trivial example. Downloading some data from the network and showing it on a screen.
+
+In UIKit-world you would:
+
+-   Create a `UIViewController` which makes the API request on its `viewDidLoad`
+-   While the request is sent you present a set of views for loading
+-   When the API request has returned you remove the loading screen
+-   You take the data from the request and create a view hierarchy then present that
+
+In React you would:
+
+-   Create a component which makes the API request on its `onMount`
+-   While the request is sent you render another component for showing loading
+-   The results come back and you change your "state" on the main component with the API request
+-   The state change re-runs your render method, which passes the API "state" down to the component for your page
+
+They are conceptually very similar. React does two key things differently: Handle "state" changes on any component, and handle view creation/addition and removal.
+
+## Handling "State" Changes
+
+So, I've been quoting "state", I should explain this. There are two types of "state" inside React, and I've been using the quoted term to refer to both for simplicity till now.
+
+> There are two types of data that control a component: `props` and `state`. `props` are set by the parent and they are fixed throughout the lifetime of a component. For data that is going to change, we have to use `state`.
+
```

Comment:
Some markdown parsers use a blank line to indicate a separate blockquote, since this is a single blockquote, I'd add `> ` to this line.